### PR TITLE
Replace `lstrip()` with `removeprefix()` to fix Ruff linter warning

### PR DESCRIPTION
**Pull request by ronensc**
This PR fixes the Ruff linter warning:
```
B005 Using `.strip()` with multi-character strings is misleading
```

I believe the intent of `data.decode("utf-8").lstrip("data:")` was to remove the prefix `data:`. However, `lstrip()` removes any combination of characters (`a`, `d`, `t`, `:`) from the beginning of the string. This could introduce potential bugs.

Ref: https://docs.python.org/3/library/stdtypes.html#str.lstrip

cc @ywang96

### CODE REVIEW

**Comment by ronensc** [comment]
@simon-mo Could you please review my PR? It's short and straightforward.

**Review by ywang96 — APPROVED** [review]
Hey @ronensc! Sorry for missing this PR but thanks for the fix!

Yes, the assumption is that every SSE token chunk is in the format of  `data:{...}` (for TGI & TRTLLM+Triton) or `data: {...}` (for OpenAI-compatible). I agree that using `removeprefix` is better and makes the intention clearer as well!

File: pyproject.toml [hunk 1]
```diff
@@ -33,8 +33,6 @@ ignore = [
     "F405", "F403",
     # lambda expression assignment
     "E731",
-    # .strip() with multi-character strings
```
Comment (rkooo567) [inline pyproject.toml:36]: do we need to delete this? Didn't this serve the purpose in this case? (that it catches this issue)

Comment (ywang96) [reply pyproject.toml:36]: This is deleting it from the `ignore` list, and I think we do want to have the ruff `B005` check on the codebase right?

**Review by simon-mo — APPROVED** [review]

**Pull request merged**